ext/java/org/msgpack/jruby/Decoder.java

```diff
@@ -140,9 +140,15 @@ private IRubyObject consumeHash(int size) {
     RubyHash hash = RubyHash.newHash(runtime);
     for (int i = 0; i < size; i++) {
       IRubyObject key = next();
-      if (this.symbolizeKeys && key instanceof RubyString) {
+      if (key instanceof RubyString) {
+        if (this.symbolizeKeys) {
           key = ((RubyString) key).intern();
+        } else {
+          key.setFrozen(true);
+          key = runtime.freezeAndDedupString((RubyString) key);
```

Comment:
Is it needed to do `key.setFrozen(true)` even before `runtime.freezeAndDedupString((RubyString) key)`?

Reply:
looking at the source, I think you're right and it's redundant https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/Ruby.java#L4830
btw I copied it from https://github.com/msgpack/msgpack-ruby/blob/master/ext/java/org/msgpack/jruby/Decoder.java#L125

Reply:
removed via https://github.com/msgpack/msgpack-ruby/pull/229/commits/a607b4496589840be7706f8bf0243cb82ecea829

Reply:
@ahorek Hey, that commit does not belong to this pull-req.